File: packages/typescript/src/tools/NavigateToUrlTool.ts
import { BaseDriver } from "../drivers/BaseDriver.ts";
import { BaseTool } from "./BaseTool.ts";
import { field, type FieldMetadata } from "./Field.ts";

export class NavigateToUrlTool extends BaseTool {
  static description = "Navigate to or open the URL.";
  static fields: FieldMetadata[] = [
    field({
      name: "url",
      type: "string",
      description: "URL to navigate to",
    }),
  ];

  url: string;

  constructor(args: { url: string }) {
    super();
    this.url = args.url;
  }

  async invoke(driver: BaseDriver): Promise<void> {
    await driver.visit(this.url);
  }
}

File: packages/python/src/alumnium/tools/navigate_to_url_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class NavigateToUrlTool(BaseTool):
    """Navigate to or open the URL."""

    url: str = Field(description="URL to navigate to")

    def invoke(self, driver: BaseDriver):
        driver.visit(self.url)
